node/src/components/consensus/highway_core/state.rs

```diff
@@ -650,7 +655,9 @@ impl<C: Context> State<C> {
         let panorama = &wunit.panorama;
         let timestamp = wunit.timestamp;
         panorama.validate(self)?;
-        if panorama.iter_correct(self).any(|v| v.timestamp > timestamp) {
+        if panorama.iter_correct(self).any(|v| v.timestamp > timestamp)
+            || wunit.timestamp < self.params.start_timestamp()
```

Comment:
Slightly faster if we do

`wunit.timestamp < self.params.start_timestamp() || panorama.iter_correct(self).any(|v| v.timestamp > timestamp)`

What changed so that now we are checking the `state_timestamp()`?

Reply:
We need to reject units with a timestamp before the start to limit the sequence numbers. (See PR description.)

Reply:
Thanks for your patience, I'll try a little better next time to answer my own questions...

Reply:
No worries! :smile: